core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -0,0 +1,229 @@
+    assertEquals(nonExistentOffsetPosition.position, cache.lookupOffset(rlsMetadata, lowerOffsetThanBaseOffset))
+  }
+
+  @Test
+  def testCacheEntryExpiry(): Unit = {
+    val cache = new RemoteIndexCache(maxSize = 2, rsm, logDir = logDir.toString)
+    val tpId = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("foo", 0))
+    val metadataList = generateRemoteLogSegmentMetadata(size = 3, tpId)
+
+    // getIndex for first time will call rsm#fetchIndex
+    cache.getIndexEntry(metadataList.head)
+    // Calling getIndex on the same entry should not call rsm#fetchIndex again, but it should retrieve from cache
+    cache.getIndexEntry(metadataList.head)
+    assertEquals(1, cache.entries.size())
+    verifyFetchIndexInvocation(count = 1)
+
+    // Here a new key metadataList(1) is invoked, that should call rsm#fetchIndex, making the count to 2
+    cache.getIndexEntry(metadataList.head)
+    cache.getIndexEntry(metadataList(1))
+    assertEquals(2, cache.entries.size())
+    verifyFetchIndexInvocation(count = 2)
+
+    // getting index for metadataList.last should call rsm#fetchIndex, but metadataList(1) is already in cache.
+    cache.getIndexEntry(metadataList.last)
+    cache.getIndexEntry(metadataList(1))
+    assertEquals(2, cache.entries.size())
+    assertTrue(cache.entries.containsKey(metadataList.last.remoteLogSegmentId().id()))
+    assertTrue(cache.entries.containsKey(metadataList(1).remoteLogSegmentId().id()))
+    verifyFetchIndexInvocation(count = 3)
+
+    // getting index for metadataList.head should call rsm#fetchIndex as that entry was expired earlier,
+    // but metadataList(1) is already in cache.
+    cache.getIndexEntry(metadataList(1))
+    cache.getIndexEntry(metadataList.head)
+    assertEquals(2, cache.entries.size())
+    assertFalse(cache.entries.containsKey(metadataList.last.remoteLogSegmentId().id()))
+    verifyFetchIndexInvocation(count = 4)
+  }
+
+  @Test
+  def testGetIndexAfterCacheClose(): Unit = {
+    val cache = new RemoteIndexCache(maxSize = 2, rsm, logDir = logDir.toString)
+    val tpId = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("foo", 0))
+    val metadataList = generateRemoteLogSegmentMetadata(size = 3, tpId)
+
+    cache.getIndexEntry(metadataList.head)
+    assertEquals(1, cache.entries.size())
+    verifyFetchIndexInvocation(count = 1)
+
+    cache.close()
+
+    // Check IllegalStateException is thrown when index is accessed after it is closed.
+    assertThrows(classOf[IllegalStateException], () => cache.getIndexEntry(metadataList.head))
+  }
+
+  @Test
+  def testReloadCacheAfterClose(): Unit = {
+    val cache = new RemoteIndexCache(maxSize = 2, rsm, logDir = logDir.toString)
+    val tpId = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("foo", 0))
+    val metadataList = generateRemoteLogSegmentMetadata(size = 3, tpId)
+
+    cache.getIndexEntry(metadataList.head)
+    cache.getIndexEntry(metadataList.head)
```

Comment:
Could we add a similar comment as in line 125?

Reply:
Done